File: Java/TestCreditCardValidator.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class TestCreditCardValidator {



	@Test
	public void testVisaCardIsValid() {
	
	String card = "4132465789041896";

	String expected = "VisaCard";

	String returned = CreditCardValidator.validateCard(card);
	
	assertEquals(expected, returned);

	}



	@Test
	public void testMasterCardIsValid() {
	
	String card = "5132465789041896";

	String expected = "MasterCard";

	String returned = CreditCardValidator.validateCard(card);
	
	assertEquals(expected, returned);

	}


	@Test
	public void testDiscoverCardIsValid() {
	
	String card = "6132465789041896";

	String expected = "Discover Card";

	String returned = CreditCardValidator.validateCard(card);
	
	assertEquals(expected, returned);

	}


	@Test
	public void testAmericanExpressCardIsValid() {
	
	String card = "371246578904189";

	String expected = "America Express";

	String returned = CreditCardValidator.validateCard(card);
	
	assertEquals(expected, returned);

	}



	@Test
	public void testLengthIsInvalid() {
	
	String card = "4132465789";

	String expected = "Invalid Card Length";

	String returned = CreditCardValidator.validateCard(card);
	
	assertEquals(expected, returned);

	}




	@Test
	public void testMasterIsValid() {
	
	String card = "5399831619690403";

	String expected = "Valid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}


	@Test
	public void testMasterIsInvalid() {
	
	String card = "5399831619690404";

	String expected = "Invalid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}



	@Test
	public void testVisaIsValid() {
	
	String card = "4388576018410707";

	String expected = "Valid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}



	@Test
	public void testVisaIsInvalid() {
	
	String card = "4388576018402626";

	String expected = "Invalid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}


	@Test
	public void testAmericanIsValid() {
	
	String card = "3788576018402629";

	String expected = "Valid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}



	@Test
	public void testAmericanIsInvalid() {
	
	String card = "3788576018402626";

	String expected = "Invalid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}



	@Test
	public void testDicoverIsValid() {
	
	String card = "6388576018410702";

	String expected = "Valid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}



	@Test
	public void testDicoverIsInvalid() {
	
	String card = "6388576018410703";

	String expected = "Invalid";

	String returned = CreditCardValidator.checkValidity(card);
	
	assertEquals(expected, returned);

	}






}

File: Python/test_credit_card_validator.py
import credit_card_validator
import unittest


class TestCardNumber(unittest.TestCase):



	def test_card_exist(self):
		credit_card_validator.validate_card


	def test_visa_card_is_valid(self):
		card = "4132465789041896"
		actual = credit_card_validator.validate_card(card)
		expected = "VisaCard"
		self.assertEqual(actual, expected)
	
	
	def test_mastercard_is_valid(self):	
		card = "5132465789041896"
		actual = credit_card_validator.validate_card(card)
		expected = "MasterCard"
		self.assertEqual(actual, expected)


	def test_discover_is_valid(self):
		card = "6132465789041896"
		actual = credit_card_validator.validate_card(card)
		expected = "DiscoverCard"
		self.assertEqual(actual, expected)



	def test_American_express_is_valid(self):	
		card = "371246578904189"
		actual = credit_card_validator.validate_card(card)
		expected = "American Express"
		self.assertEqual(actual, expected)



	def test_length_is_invalid(self):	
		card = "4132465789"
		actual = credit_card_validator.validate_card(card)
		expected = "Invalid Card Length"
		self.assertEqual(actual, expected)


	def test_mastercard_is_valid_pass(self):	
		card = "5399831619690403"
		actual = credit_card_validator.check_validity(card)
		expected = "Valid"
		self.assertEqual(actual, expected)


	def test_mastercard_is_invalid(self):	
		card = "5399831619690404"
		actual = credit_card_validator.check_validity(card)
		expected = "Invalid"
		self.assertEqual(actual, expected)


	def test_visacard_is_valid(self):	
		card = "4388576018410707"
		actual = credit_card_validator.check_validity(card)
		expected = "Valid"
		self.assertEqual(actual, expected)


	def test_visacard_is_invalid(self):	
		card = "4388576018402626"
		actual = credit_card_validator.check_validity(card)
		expected = "Invalid"
		self.assertEqual(actual, expected)


	def test_americanexpress_is_invalid(self):	
		card = "3788576018402626"
		actual = credit_card_validator.check_validity(card)
		expected = "Invalid"
		self.assertEqual(actual, expected)


	def test_americanexpress_is_valid(self):	
		card = "3788576018402629"
		actual = credit_card_validator.check_validity(card)
		expected = "Valid"
		self.assertEqual(actual, expected)


	def test_discover_is_valid(self):
		card = "6388576018410702"
		actual = credit_card_validator.check_validity(card)
		expected = "Valid"
		self.assertEqual(actual, expected)


	def test_discover_is_invalid(self):
		card = "6388576018410703"
		actual = credit_card_validator.check_validity(card)
		expected = "Invalid"
		self.assertEqual(actual, expected)














